include/ListView/ListViewSmarty.php

```diff
@@ -79,14 +79,51 @@ public function __construct() {
         parent::__construct();
         $this->ss = new Sugar_Smarty();
     }
+            
+    /**
+     * 
+     * @global array $app_strings
+     * @global array $sugar_config
+     * @return string|boolean
+     */
+    public function buildSendConfirmOptInEmailToPersonAndCompany() {
+        
+        global $app_strings, $sugar_config;
+        
+        $confirmOptInEnabled = isset($sugar_config['email_enable_confirm_opt_in']) && $sugar_config['email_enable_confirm_opt_in'];
+
+        if (!$confirmOptInEnabled) {
+            $this->warn('Confirm Opt In disabled');
+            return false;
+        }
+        
+        $js = "sListView.send_form("
```

Comment:
Are we not trying to move away from hard coded content in PHP?

Could we not move this into the template some how?

Reply:
outsourced